File: app/src/main/java/com/jerson/hcdc_portal/ui/adapter/AccountAdapter.java
package com.jerson.hcdc_portal.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.jerson.hcdc_portal.databinding.ItemContainerAccountBinding;
import com.jerson.hcdc_portal.databinding.ItemContainerGradeBinding;
import com.jerson.hcdc_portal.model.AccountModel;

import java.util.List;

public class AccountAdapter extends RecyclerView.Adapter<AccountAdapter.ViewHolder> {

    Context context;
    List<AccountModel> list;

    public AccountAdapter(Context context, List<AccountModel> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public AccountAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ItemContainerAccountBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull AccountAdapter.ViewHolder holder, int position) {
        AccountModel model = list.get(position);

        holder.binding.refNo.setText(model.getReference());
        holder.binding.added.setText(model.getAdded());
        holder.binding.deducted.setText(model.getAdded());
        holder.binding.runBal.setText(model.getRunBal());
        holder.binding.date.setText(model.getDate());
        holder.binding.desc.setText(model.getDescription());

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ItemContainerAccountBinding binding;
        public ViewHolder(ItemContainerAccountBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/jerson/hcdc_portal/presentation/accounts/adapter/AccountAdapter.kt
package com.jerson.hcdc_portal.presentation.accounts.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.jerson.hcdc_portal.databinding.ItemAccountBinding
import com.jerson.hcdc_portal.domain.model.Account

class AccountAdapter(private val list:List<Account>):RecyclerView.Adapter<AccountAdapter.ViewHolder>() {

    inner class ViewHolder(private val binding:ItemAccountBinding):RecyclerView.ViewHolder(binding.root){
        fun bind(item:Account){
            binding.apply{
                date.text = item.date
                refNo.text = item.reference
                added.text = item.added
                deducted.text = item.deducted
                runBal.text = item.runningBal
                desc.text = item.description
            }
        }
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AccountAdapter.ViewHolder {
        return ViewHolder(
            ItemAccountBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: AccountAdapter.ViewHolder, position: Int) {
        val item = list[position]
        holder.bind(item)
    }

    override fun getItemCount(): Int {
        return list.size
    }

}